### PR TITLE
chore: rename night mode to dark theme

### PR DESCRIPTION
**Pull request by Udit-takkar**
#5169 
Renamed all the files to dark theme .

Last inactive open PR https://github.com/zulip/zulip-mobile/pull/5170.
I have run all the tests and also changed some more files that were not addressed in last PR

### CODE REVIEW

**Comment by Udit-takkar** [comment]
It is ready to be reviewed @chrisbobbe

**Comment by chrisbobbe** [comment]
`src/api/initialDataTypes.js` has our [Flow types](https://flow.org/) that represent the data we expect to receive from servers. We should only change those types if

- they've been doing their job badly (they were written without understanding what servers are actually sending)
- new servers have started sending different data, and we want to start representing the new data.

I don't think either of those applies here. (If that's wrong, please talk about why you think so. 🙂) So we should leave those types unchanged, and Flow will lead you to undo several follow-on changes that made it into this PR. Thanks!

**Comment by Udit-takkar** [comment]
> `src/api/initialDataTypes.js` has our [Flow types](https://flow.org/) that represent the data we expect to receive from servers. We should only change those types if
> 
> * they've been doing their job badly (they were written without understanding what servers are actually sending)
> * new servers have started sending different data, and we want to start representing the new data.
> 
> I don't think either of those applies here. (If that's wrong, please talk about why you think so. 🙂) So we should leave those types unchanged, and Flow will lead you to undo several follow-on changes you made. Thanks!

Okay you are right

**Comment by Udit-takkar** [comment]
@chrisbobbe  is it fine now?

File: docs/changelog.md [hunk 1]
```diff
@@ -1394,7 +1394,7 @@ Bugfixes and other improvements for your Zulip experience.
 ### Highlights for users (since 26.1.124 / 26.2.125)
 
 * Highlight colors for code blocks now match the webapp and
-  offer more contrast, especially in night mode.
+  offer more contrast, especially in dark theme.
```
Comment (gnprice) [inline docs/changelog.md:1397]: This is a historical changelog entry (from 2019), so let's leave it in the form we used at the time.

File: src/start/IosCompliantAppleAuthButton/Custom.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const styles = createStyleSheet({
     borderRadius: 22,
     overflow: 'hidden',
   },
-  nightFrame: {
+  darkFrame: {
     backgroundColor: 'black',
   },
   dayFrame: {
```
Comment (gnprice) [inline src/start/IosCompliantAppleAuthButton/Custom.js:29]: This should change from "day" to "light", to match the change from "night" to "dark".

Same for the other mention of "day" in this file.

File: src/webview/css/css.js [hunk 3]
```diff
@@ -26,8 +26,8 @@ export default (theme: ThemeName): string => `
 <link rel='stylesheet' type='text/css' href='./base.css'>
 <link rel='stylesheet' type='text/css' href='./katex/katex.min.css'>
 <style>
-${theme === 'night' ? cssNight : ''}
-${cssPygments(theme === 'night')}
+${theme === 'dark' ? cssNight : ''}
```
Comment (gnprice) [inline src/webview/css/css.js:29]: That file `cssNight.js` should be renamed too.

**Review by gnprice — COMMENTED** [review]
Thanks @Udit-takkar! Some comments below.

Also please take a look at the Zulip project's guide on how to make a clean commit history:
https://zulip.readthedocs.io/en/latest/contributing/version-control.html

In particular, rather than having one commit with a bug and a commit later in the PR that fixes it, we ask that you squash the fix to make one commit that doesn't have the bug. So in this PR, your three commits so far should be squashed together as one commit.

File: docs/changelog.md (hunk 1 above)
Comment (Udit-takkar) [reply docs/changelog.md:1397]: Okay

**Comment by Udit-takkar** [comment]
Okay @gnprice  did all the changes you suggested and cleaned the git history

**Pull request closed without merging**